### PR TITLE
feat: added inject migration

### PR DESCRIPTION
**Pull request by eneajaho**
It supports everything 🎉

Before: 
test (1)

After: 
test (2)

Also inspiration taken from: https://github.com/kreuzerk/inject-pilot/tree/main 

cc @kreuzerk @ilirbeqirii

### CODE REVIEW

File: libs/plugin/src/generators/convert-di-to-inject/generator.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+				constructor.getParameters().forEach((param) => {
+					if (convertedDeps.has(param.getName())) {
+						param.remove();
+					}
+				});
+
+				if (
+					constructor.getParameters().length === 0 &&
+					constructor.getBodyText().trim() === ''
```
Comment (scr2em) [inline libs/plugin/src/generators/convert-di-to-inject/generator.ts:225]: @eneajaho  I think you should check if the constructor has `super()` only, if that's the case the constructor should be removed as well 
```
constructor.getBodyText().trim() === 'super()'
```

Comment (ilirbeqirii) [reply libs/plugin/src/generators/convert-di-to-inject/generator.ts:225]: I think we should check first if the schematic converts also the DI injections in the plain classes - ones being decorated anyway. if yes, then we can also remove the constructors with super() only. I guess 👀

**Comment by k3nsei** [comment]
Should MyService2 not be public property in class instance? As it’s just argument which should be just in constructor scope.

```typescript
constructor() {
  const myService2 = inject(MyService2);
}
```

**Comment by eneajaho** [comment]
> Shouldn’t MyService2 be just const in constructor scope and not public property in class instance?
> 
> ```ts
> constructor() {
>   const myService2 = inject(MyService2);
> }
> ```

Yeah, you're right about this one. 

Now I'm not sure on what I should do with this. Should we unify the deps or keep it inside the constructor only for that one case?

**Comment by ilirbeqirii** [comment]
> > Shouldn’t MyService2 be just const in constructor scope and not public property in class instance?
> > ```ts
> > constructor() {
> >   const myService2 = inject(MyService2);
> > }
> > ```
> 
> Yeah, you're right about this one.
> 
> Now I'm not sure on what I should do with this. Should we unify the deps or keep it inside the constructor only for that one case?

man, I think you should keep it inside the constructor as that's its scope of work.

**Comment by eneajaho** [comment]
> > > Shouldn’t MyService2 be just const in constructor scope and not public property in class instance?
> > > ```ts
> > > constructor() {
> > >   const myService2 = inject(MyService2);
> > > }
> > > ```
> > 
> > 
> > Yeah, you're right about this one.
> > Now I'm not sure on what I should do with this. Should we unify the deps or keep it inside the constructor only for that one case?
> 
> man, I think you should keep it inside the constructor as that's its scope of work.

Most of the time I've seen this be used as: 

```ts
class Cmp {
  myService2: MyService2;

  constructor(myService2: MyService2;) {
     // just re-assign to a class field
     this.myService2 = myService2;
  }

}
```

So, I'm keep the same behavior by creating it as a variable inside the constructor.

**Pull request merged**